cmd/fleet/serve.go

```diff
@@ -708,6 +708,12 @@ func cronVulnerabilities(
 			continue
 		}
 
+		if err := ds.CalculateHostsPerSoftware(ctx, time.Now()); err != nil {
```

Comment:
Please remind me, users cannot have software-inventory=ON and vulnerabilities=OFF, right? 
(Asking because I would expect this feature to work in such scenario)

Reply:
I'm not totally sure based on this: https://github.com/fleetdm/fleet/blob/main/docs/01-Using-Fleet/13-Vulnerability-Processing.md#configuration ? Seems like they're both enabled by default, but you can still have software inventory but not vulnerability processing? If so, this would indeed be a problem, unless we're ok with not having counts when only software inventory is enabled (which might be fine, as it looks like the way counts are used is related to vulnerabilities investigation, but definitely worth double-checking! The goal of the ticket is `As a **vulnerability analyst** with a lot of hosts, I want a count of hosts for each software item to be included so I understand **how many hosts are impacted by a vulnerability** (CVE).`). / @chiiph @noahtalerman

Reply:
We plan on having only one switch for both features, but for now you can have only software inventory enabled.

We could move the databases path check within the for loop and only wrap the vulnerability checks with it so that the counts are still calculated as long as software inventory is enabled. What do you think?

Reply:
> We could move the databases path check within the for loop and only wrap the vulnerability checks with it so that the counts are still calculated as long as software inventory is enabled. What do you think?

That could work, but this means that by default, software inventory of hosts will "show up" after ~1h (`/api/v1/fleet/software` will return no hosts if the counts haven't been set yet.)

Reply:
Yeah I think it'd be simple enough, I'll update the PR.

Reply:
Note that it'd still require `current_instance_checks` to be at least auto/on, not off on all instances.

Reply:
> software inventory of hosts will "show up" after ~1h (/api/v1/fleet/software will return no hosts if the counts haven't been set yet.)

AFAICT this is by design: https://github.com/fleetdm/fleet/issues/3086#issuecomment-1014844143

Reply:
Oh! I totally missed that comment, if that's expected then we are g2g.

I think we missed adding that information on the issue description (mostly the UX change: now users have to wait for ~1h or so until they see listed software).

Reply:
Updated, if y'all want to take a look: https://github.com/fleetdm/fleet/pull/3873/files#diff-92cebe996f3c912cf7faa9227ee3fa22fc6b6d225f6628bb6b9c277b54e1dbb3

(as mentioned, this still requires the `current_instance_checks` to be either `on` or `auto`).